src/components/document-viewer/DocFragment.jsx

```diff
@@ -22,26 +25,86 @@ class DocFragment extends Component {
   constructor(props) {
     super(props);
 
+    const json = {};
+    json[this.props.title] = _.pickBy(
```

Comment:
Looks like you could can wrap this is a `setTitle` function as it's used again later.